runtime/src/qio/auxFilesys/hdfs/qio_plugin_hdfs.c

```diff
@@ -493,10 +493,15 @@ const qio_file_functions_ptr_t hdfs_function_struct_ptr = &hdfs_function_struct;
 static
 char* get_locale_name(char *loc)
 {
+  char* ret = NULL;
   const char *keys = "."; // We lop off the .us.cray...
-  int i = strcspn (loc, keys);
-
-  return strndup(loc, i);
+  int len = strcspn (loc, keys);
+  ret = (char*) qio_malloc(len+1);
+  if (ret) {
+    ret[len] = '\0';
+    qio_memcpy(ret, loc, len);
```

Comment:
I'm happy to change, and I slightly prefer the ordering you suggest (easy to read/follow IMO). I went with the current approach because I was looking at the glibc strndup and figured they probably have a reason for putting '\0' first: https://code.woboq.org/userspace/glibc/string/strndup.c.html#48